napari/utils/translations.py

```diff
@@ -216,160 +305,183 @@ def _update_locale(self, locale: str):
+        kwargs : dict, optional
+            Any additional arguments to use when formating the string.
 
         Returns
         -------
-        str
-            The translated string.
+        TranslationString or str
+            The translation string which might be deferred or translated in
+            place.
         """
-        # Python 3.7 or lower does not offer translations based on context.
-        # On these versions `gettext.pgettext` falls back to `gettext.gettext`
-        if PY37_OR_LOWER:
-            translation = gettext.dgettext(self._domain, msgid)
+        if deferred:
+            translation = TranslationString(
+                domain=self._domain, msgid=msgid, deferred=deferred, **kwargs
+            )
         else:
-            translation = gettext.dpgettext(self._domain, msgctxt, msgid)
+            translation = self._dnpgettext(msgid=msgid, **kwargs)
 
         return translation
 
-    def npgettext(
-        self, msgctxt: str, msgid: str, msgid_plural: str, n: int
-    ) -> str:
+    def _n(
+        self,
+        msgid: str,
+        msgid_plural: str,
+        n: int,
+        deferred: bool = False,
+        **kwargs,
+    ) -> Union[TranslationString, str]:
         """
-        Translate a singular string with context and pluralization.
+        Shorthand for `gettext.ngettext` with enhanced functionality.
 
         Parameters
         ----------
-        msgctxt : str
-            The message context.
         msgid : str
             The singular string to translate.
         msgid_plural : str
             The plural string to translate.
         n : int
             The number for pluralization.
+        deferred : bool, optional
+            Define if the string translation should be deferred or executed
+            in place. Default is False.
+        kwargs : dict, optional
+            Any additional arguments to use when formating the string.
 
         Returns
         -------
-        str
-            The translated string.
+        TranslationString or str
+            The translation string which might be deferred or translated in
+            place.
         """
-        # Python 3.7 or lower does not offer translations based on context.
-        # On these versions `gettext.npgettext` falls back to `gettext.ngettext`
-        if PY37_OR_LOWER:
-            translation = gettext.dngettext(
-                self._domain, msgid, msgid_plural, n
+        if deferred:
+            translation = TranslationString(
+                domain=self._domain,
+                msgid=msgid,
+                msgid_plural=msgid_plural,
+                n=n,
+                deferred=deferred,
+                **kwargs,
             )
         else:
-            translation = gettext.dnpgettext(
-                self._domain, msgctxt, msgid, msgid_plural, n
+            translation = self._dnpgettext(
+                msgid=msgid, msgid_plural=msgid_plural, n=n, **kwargs
             )
```

Comment:
do not use `translation` variable. Use `return` statement and remove obsolete `else`.